src/main/groovy/grails/plugins/mail/graph/sender/GraphMailSenderImpl.groovy

```diff
@@ -35,43 +35,51 @@ class GraphMailSenderImpl extends OAuthMailSenderImpl {
 
     @Override
     protected void doSend(MimeMessage[] mimeMessages, Object[] originalMessages) throws MailException {
-        log.warn("Invalid method of GraphMailSenderImpl getting called")
-        throw new GrailsMailException("Please do use sendMailViaGraph method rather doSend")
+        log.warn("[GRAPH_EMAIL] [UNSUPPORTED_METHOD] doSend() called — This method is not supported, please use sendMailViaGraph.")
+        throw new GrailsMailException("Please use sendMailViaGraph instead of doSend.")
     }
 
     void sendMailViaGraph(Message message, List<FileAttachment> attachmentList) throws MailException {
-        Map<Object, Exception> failedMessages = new LinkedHashMap<Object, Exception>()
+        Map<Object, Exception> failedMessages = new LinkedHashMap<>()
         boolean connectionStatus = mailOAuthService.accessToken
+
         try {
             if (!connectionStatus) {
                 throw new MailAuthenticationException(new AuthenticationFailedException())
             }
         } catch (Exception ex) {
+            log.error("[GRAPH_EMAIL] [AUTH_FAILURE] Failed to get access token: ${ex.message}", ex)
             failedMessages.put(message, ex)
             throw new MailSendException("Mail server connection failed", ex, failedMessages)
         }
+
         try {
+            log.info("[GRAPH_EMAIL] [SEND_EMAIL] [STARTED] - Recipients=${message?.toRecipients*.emailAddress*.address} - From=${message?.from.emailAddress.address} | Attachments=${attachmentList?.size()}")
```

Comment:
@RxL-Farhan-Ali Keep it at debug level not info. If any application require then it can open debug level.